### PR TITLE
QemuHCK: Do not add existing values from states to array

### PR DESCRIPTION
**Pull request by kostyanf14**

### CODE REVIEW

File: lib/setupmanagers/qemuhck/qemu_machine.rb [hunk 1]
```diff
@@ -300,7 +300,7 @@
         when Hash
           var_value.merge! value
         when Array
-          var_value << value unless var_value.include? value
+          var_value |= value
```
[the remark this answers is not in the document]
Comment (kostyanf14) [reply lib/setupmanagers/qemuhck/qemu_machine.rb:303]: @akihikodaki I think this is false-positive, `var_value` is not a local variable in our case.

[the remark this answers is not in the document]
Comment (akihikodaki) [reply lib/setupmanagers/qemuhck/qemu_machine.rb:303]: This is not a false positive; you need to `instance_variable_set`.

[the remark this answers is not in the document]
Comment (kostyanf14) [reply lib/setupmanagers/qemuhck/qemu_machine.rb:303]: @akihikodaki Fixed

File: lib/setupmanagers/qemuhck/qemu_machine.rb [hunk 2]
```diff
@@ -220,7 +220,7 @@ def define_local_variables
       @machine_extra_param = []
       @device_extra_param = []
       @iommu_device_param = []
-      @cpu_options = []
+      @cpu_options = ['@cpu@', '+x2apic', '+fsgsbase', 'model=@cpu_model@']
```
Comment (akihikodaki) [inline lib/setupmanagers/qemuhck/qemu_machine.rb:223]: I think it's better not to put `@cpu@` here; `@cpu@` is different from other options in terms that it must be present before the other options.

Also, here is some Ruby magic:
```suggestion
      @cpu_options = %w[@cpu@ +x2apic +fsgsbase model=@cpu_model@]
```

File: lib/setupmanagers/qemuhck/states.json [hunk 3]
```diff
@@ -49,7 +54,9 @@
     },
     "drive_unsafe_cache_state": {
         "true": {
-            "drive_cache_options": ",cache=unsafe"
+            "drive_cache_options": [
+                ",cache=unsafe"
```
Comment (akihikodaki) [inline lib/setupmanagers/qemuhck/states.json:58]:
```suggestion
                "cache=unsafe"
```

**Review by akihikodaki — CHANGES_REQUESTED** [review]

File: lib/setupmanagers/qemuhck/states.json (hunk 3 above)
Comment (kostyanf14) [reply lib/setupmanagers/qemuhck/states.json:58]: Unfortunately, no. We use `@drive_cache_options@` in `"-drive file=@image_path@,if=none,id=ide_@run_id@_@client_id@@drive_cache_options@"` and should somehow add `,` before drive_cache_options if it present.

Comment (kostyanf14) [reply lib/setupmanagers/qemuhck/states.json:58]: In any case we need some if/map/etc to add ',' before all options but only if options is not empty

**Review by akihikodaki — APPROVED** [review]

**Pull request merged**